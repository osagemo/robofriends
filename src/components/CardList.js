import React, { Fragment } from "react";
import Card from "./Card";

export default function CardList({ robots }) {
  return (
    <Fragment>
      {robots.map(robot => (
        <Card
          key={robot.id}
          id={robot.id}
          name={robot.name}
          email={robot.email}
        />
      ))}
    </Fragment>
  );
}
