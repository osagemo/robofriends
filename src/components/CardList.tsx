import React, { Fragment } from "react";
import Card from "./Card";
import { IRobot } from "../types";

interface IMainPageProps {
  robots: Array<IRobot>;
}

interface IMainPageState {}

export default function CardList({ robots }: IMainPageProps) {
  return (
    <Fragment>
      {robots.map((robot: IRobot) => (
        <Card
          key={robot.id}
          id={robot.id}
          name={robot.name}
          email={robot.email}
        />
      ))}
    </Fragment>
  );
}
