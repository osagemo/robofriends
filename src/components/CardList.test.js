import { shallow } from "enzyme";
import React from "react";
import CardList from "./CardList";

it("expect to render CardList component with mock users", () => {
  const robotsMock = [
    {
      id: 1,
      name: "Kalle King",
      email: "abc@abc.se"
    },
    {
      id: 2,
      name: "Apkungen",
      email: "mk@gårpåtå.nu"
    }
  ];
  const wrapper = shallow(<CardList robots={robotsMock} />);
  expect(wrapper).toMatchSnapshot();
});
