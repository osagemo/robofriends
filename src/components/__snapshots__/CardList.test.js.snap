// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component with mock users 1`] = `
<Fragment>
  <Card
    email="abc@abc.se"
    id={1}
    key="1"
    name="Kalle King"
  />
  <Card
    email="mk@gårpåtå.nu"
    id={2}
    key="2"
    name="Apkungen"
  />
</Fragment>
`;
